chromadb/db/__init__.py

```diff
@@ -1,125 +1,155 @@
-    ):
-        pass
+class Segment(TypedDict):
+    id: UUID
+    python_class: str
+    metadata: StrDict
 
-    @abstractmethod
-    def delete_collection(self, name: str):
-        pass
 
-    @abstractmethod
-    def get_collection_uuid_from_name(self, collection_name: str) -> str:
-        pass
+class System(ABC):
+    """Interface for Chroma's System storage backend"""
+
 
     @abstractmethod
-    def add(
-        self,
-        collection_uuid: str,
-        embeddings: Embeddings,
-        metadatas: Optional[Metadatas],
-        documents: Optional[Documents],
-        ids: List[UUID],
-    ) -> List[UUID]:
-        pass
+    def create_collection(self,
+                          name: str,
+                          metadata: StrDict = None,
+                          get_or_create: bool = False) -> Collection:
+       """Create a new collection"""
+       pass
+
 
     @abstractmethod
-    def add_incremental(self, collection_uuid: str, ids: List[UUID], embeddings: Embeddings):
+    def get_collection(self,
+                       name_or_id: NameOrID) -> Collection:
+        """Get a collection"""
         pass
 
+
     @abstractmethod
-    def get(
-        self,
-        where: Where = {},
-        collection_name: Optional[str] = None,
-        collection_uuid: Optional[str] = None,
-        ids: Optional[IDs] = None,
-        sort: Optional[str] = None,
-        limit: Optional[int] = None,
-        offset: Optional[int] = None,
-        where_document: WhereDocument = {},
-        columns: Optional[List[str]] = None,
-    ) -> Sequence:
+    def list_collections(self) -> Sequence[Collection]:
+        """List all collections"""
         pass
 
+
     @abstractmethod
-    def update(
-        self,
-        collection_uuid: str,
-        ids: IDs,
-        embeddings: Optional[Embeddings] = None,
-        metadatas: Optional[Metadatas] = None,
-        documents: Optional[Documents] = None,
-    ):
+    def update_collection(self,
+                          name_or_id: NameOrID,
+                          new_name: Optional[str] = None,
+                          new_metadata: StrDict = None) -> Collection:
+        """Update a collection's name and/or metadata"""
         pass
 
+
     @abstractmethod
-    def count(self, collection_name: str):
+    def delete_collection(self,  name_or_id: NameOrID) -> None:
+        """Delete a collection by name or id. Returns None on success."""
         pass
 
+
+    # TODO: This method signature needs more thought put into it, and will likely
+    # need to change or be split into multiple methods as the system grows.
     @abstractmethod
-    def delete(
-        self,
-        where: Where = {},
-        collection_name: Optional[str] = None,
-        collection_uuid: Optional[str] = None,
-        ids: Optional[IDs] = None,
-        where_document: WhereDocument = {},
-    ):
+    def get_segment(self,
```

Comment:
should create_segment be exposed?

Reply:
Ok sounds good - I am asking because I want to understand what affordances the placement engine I'm thinking about will have.